library/pk.c

```diff
@@ -175,7 +184,6 @@ int mbedtls_pk_setup_rsa_alt( mbedtls_pk_context *ctx, void * key,
  */
 int mbedtls_pk_can_do( const mbedtls_pk_context *ctx, mbedtls_pk_type_t type )
 {
-    /* null or NONE context can't do anything */
     if( ctx == NULL || ctx->pk_info == NULL )
```

Comment:
This check should be moved to parameter validation.

Reply:
`can_do` is a function that one might plausibly call early on before bailing out because allocating the context failed, and returning 0 here is safe, so I prefer to err on the side of backward compatibility. We can officially deprecate it later and change the behavior after that.

Reply:
Fine by me.